rust/compaction/src/lib.rs

```diff
@@ -85,6 +85,7 @@ pub struct FFICompactionParams {
     aws_endpoint: *const c_char,
     aws_access_key: *const c_char,
     aws_secret_key: *const c_char,
+    aws_allow_http: bool,
```

Comment:
The field order is different from what is defined on the Java side. This will cause problems when accessing the memory from Rust.

Reply:
That's fixed, thanks.